src/sentry/monitors/logic/mark_ok.py

```diff
@@ -33,17 +21,49 @@ def mark_ok(checkin: MonitorCheckIn, ts: datetime):
         "next_checkin": next_checkin,
         "next_checkin_latest": next_checkin_latest,
     }
-    if checkin.status == CheckInStatus.OK:
+
+    # If the monitor is already in OK status there's not incident update or
+    # status change necessary
+    if monitor_env.status == CheckInStatus.OK:
+        MonitorEnvironment.objects.filter(id=monitor_env.id).exclude(last_checkin__gt=ts).update(
+            **params
+        )
+        return
+
+    recovery_threshold = monitor_env.monitor.config.get("recovery_threshold", 0)
+    using_incidents = bool(recovery_threshold)
```

Comment:
maybe just change this to `recovery_threshold > 1` which is functionally the same thing for now

Reply:
Yeah I basically just duplicated what the logic was before `if recovery_threshold:`